boa/src/syntax/parser/statement/iteration/do_while_statement.rs

```diff
@@ -63,11 +63,20 @@ where
 
     fn parse(self, cursor: &mut Cursor<R>) -> Result<Self::Output, ParseError> {
         let _timer = BoaProfiler::global().start_event("DoWhileStatement", "Parsing");
-        cursor.expect(Keyword::Do, "do while statement")?;
+
+        let position = cursor
+            .expect(Keyword::Do, "do while statement")?
+            .span()
+            .end();
 
         let body =
             Statement::new(self.allow_yield, self.allow_await, self.allow_return).parse(cursor)?;
 
+        // Early Error: It is a Syntax Error if IsLabelledFunction(Statement) is true.
+        if let Node::FunctionDecl(_) = body {
+            return Err(ParseError::general("In non-strict mode code, functions can only be declared at top level, inside a block, or as the body of an if statement.", position));
```

Comment:
Could you extract this error message as a const `&str`?

Reply:
Moved it to a method on `ParseError` to avoid importing.